lib/codec_choose.c

```diff
@@ -49,6 +49,9 @@
 #ifndef bit_SSE42
 #define bit_SSE42 (1 << 20)
 #endif
+#ifndef bit_AVX
+#define bit_AVX (1 << 28)
+#endif
 
 #define bit_XSAVE_XRSTORE (1 << 27)
```

Comment:
I know this is not your code, but I was looking at the [Wikipedia page for CPUID](https://en.wikipedia.org/wiki/CPUID) and it looks like it might be better to use bit 26 instead of 27 to check for AVX. Bit 26 is whether the hardware supports `XSAVE` and `XRESTOR`, bit 27 is whether the OS supports `XRESTOR`. If you agree, this commit would be the place to change it. Compiling with bit 26 works fine for me. (Question: how can an OS influence the result of `CPUID`?)